lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -367,6 +367,12 @@ def install_puppet_agent_on(hosts, opts = {})
             end
 
             if package_name
+              # Some SLES images already have rubygem-puppet installed and this
```

Comment:
What SLES images are you dealing with that have rubygem-puppet installed? That sounds like that should be dealt with at an image level or within the pre-suite of your testing. I don't think that adding that here is the appropriate place for this logic.